presto-hive/src/test/java/com/facebook/presto/hive/TestShowStats.java

```diff
@@ -104,6 +112,22 @@ public void testShowStats()
                         "   ('name', 37.0, 5.0, 0.0, null, null, null), " +
                         "   ('comment', 310.0, 5.0, 0.0, null, null, null), " +
                         "   (null, null, null, null, 5.0, null, null))");
+
+        assertQuery("SHOW STATS FOR (SELECT * FROM nation_partitioned WHERE sin(regionkey) > 1)",
```

Comment:
How does this work? Is there some clever logic that detects that `sin` is never `> 1`?

Reply:
I didn't find anything special for sin(), my best guess it that pushing down may have some CBO to evaluate expressions.

Reply:
@mbasmanova It turns out that sin(regionkey) is calculated and this predicate is evaluated for each partition key in test dataset. For the case of "sin(regionkey) > 1",  no partition satisfy the condition. Note that sin() function use radian as unit.  Thanks for tip from @rschlussel

Reply:
@picrinite Vic, that makes sense, but doesn't explain why `SHOW STATS FOR (SELECT * FROM nation_partitioned WHERE sin(regionkey) > 0)` fails with `Only predicates that can be pushed down are supported in the SHOW STATS WHERE clause`.